File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int open_cnt = 0;
        int ans = 0;

        for (char c: s) {
            if (c == '(') {
                open_cnt++;
            } else {
                open_cnt > 0 ? open_cnt-- : ans++;
            }
        }
        return open_cnt + ans; 
    }
};

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        ans = 0
        open_cnt = 0
        for c in s:
            if c == '(':
                open_cnt += 1
            elif c == ')':
                if open_cnt > 0:
                    open_cnt -= 1
                elif open_cnt == 0:
                    ans += 1
        return ans + open_cnt